server/grpc_test.go

```diff
@@ -410,6 +412,65 @@ func TestGrpcAcRequestInlinedBlobs(t *testing.T) {
 	}
 }
 
+func TestGrpcByteStreamDeadline(t *testing.T) {
+	testBlobSize := int64(maxChunkSize * 3 / 2)
+	testBlob, testBlobHash := testutils.RandomDataAndHash(testBlobSize)
+	testBlobDigest := pb.Digest{
+		Hash:      testBlobHash,
+		SizeBytes: int64(len(testBlob)),
+	}
+
+	instance := "deadlineExpired"
+
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(10*time.Millisecond))
+	defer cancel()
+
+	bswc, err := bsClient.Write(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	resourceName := fmt.Sprintf("%s/uploads/%s/blobs/%s/%d/deadline/metadata/here", instance, uuid.New().String(), testBlobDigest.Hash, len(testBlob))
+	fmt.Println(testBlobHash)
+
+	for i := 0; i < len(testBlob); i++ {
+		bswReq := bytestream.WriteRequest{
+			ResourceName: resourceName,
+			FinishWrite:  false,
+			Data:         testBlob[i : i+1],
+			WriteOffset:  int64(i),
+		}
+
+		err := bswc.Send(&bswReq)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			t.Error(err)
+		}
+
+		time.Sleep(time.Millisecond)
+	}
+
+	_, err = bswc.CloseAndRecv()
+
+	statusError, ok := status.FromError(err)
+	if !ok {
+		t.Fatalf("expected rpc error code, got %v\n", err)
+	}
+
+	if code := statusError.Code(); code != codes.DeadlineExceeded {
+		t.Errorf("expected codes.DeadlineExceeded, got %s\n", code.String())
```

Comment:
Should this be `t.Fatalf(...)` ?

Reply:
I don't know if it makes much of a difference, I usually reserve fatal for failed assertions which would later panic